code/app/src/main/java/dev/tonholo/study/pokedex/screens/pokemonList/PokemonListViewModel.kt

```diff
@@ -40,45 +47,100 @@ class PokemonListViewModel @Inject constructor(
     }
 
     fun loadPokemonListPaginated() {
+        val offset = PAGE_SIZE
+        val cursor = currentPage * PAGE_SIZE
+        Log.d(TAG, "loadPokemonListPaginated: currentPage = $currentPage")
+        Log.d(TAG, "loadPokemonListPaginated: pokemonList.size = ${pokemonList.value.size}")
+        Log.d(TAG, "loadPokemonListPaginated: isEndReached = $isEndReached")
+
         viewModelScope.launch {
-            isLoading.value = true
-            val params = GetPokemonListUseCase.Params(
-                limit = PAGE_SIZE,
-                offset = currentPage * PAGE_SIZE,
+            networkBoundResource(
```

Comment:
I know you closed this MR already but I just would like to give my 2 cents as you asked before 😄 

This kind of reusable function for this MR that was something that caught my attention.

Comparing to the https://github.com/rafaeltonholo/study-pokedex/pull/3, this function was harder to understand at first. So I would say that the https://github.com/rafaeltonholo/study-pokedex/pull/3 is better straight forward to understand.

But, this one have better reusability as well right, because of this function. But IMHO, I would prefer a code better to understand on this case (https://github.com/rafaeltonholo/study-pokedex/pull/3).